### PR TITLE
issue #100 - completed sha3. added a unit test for sha3

### PR DESCRIPTION
**Pull request by r-soltani**
This issue adds a unit test for sha3. It also updates the SHA3 class with a finalize method.

### CODE REVIEW

File: src/eqb/sha3/sha3.h [hunk 1]
```diff
@@ -26,14 +26,18 @@ class SHA3
 public:
     static const size_t OUTPUT_SIZE = 32;
 
+    SHA3() : m_type(0), m_version(0)
```
Comment (macterra) [inline src/eqb/sha3/sha3.h]: What are type and version used for?

File: src/eqb/sha3/sha3.h [hunk 2]
```diff
@@ -44,15 +48,14 @@ class SHA3
 
     void Finalize(unsigned char hash[OUTPUT_SIZE])
     {
-    	// EQB_TODO: Implement Finalize
+        uint256 hashresult = GetHash();
+        memcpy(hash, (unsigned char*) &hashresult, OUTPUT_SIZE);
     }
```
Comment (macterra) [inline src/eqb/sha3/sha3.h]: Needs a Reset() method:

```

    SHA3& Reset()
    {
        m_serialization.clear();
        return *this;
    }
```

Comment (r-soltani) [reply src/eqb/sha3/sha3.h]: Done

File: src/eqb/sha3/sha3.h (hunk 1 above)
Comment (r-soltani) [reply src/eqb/sha3/sha3.h]: Removed the version and type for this version until later. I added those to support variations of SHA3.

**Review by macterra — APPROVED** [review]

**Review by ealefad — APPROVED** [review]

**Pull request merged**